lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -386,6 +419,28 @@ Parser.prototype.onSettingsFrame = function onSettingsFrame(header,
       settings[name] = value;
   }
 
+  if (constants.settingsIndex[constants.settings_server.SETTINGS_ENABLE_PUSH] in settings) {
```

Comment:
Woooow, this looks a bit long. Any way it could be shrunk with all repeating code removed?